librz/io/p/io_ar.c

```diff
@@ -12,19 +12,32 @@ static bool rz_io_ar_plugin_open(RzIO *io, const char *file, bool many) {
 
 static RzIODesc *rz_io_ar_open(RzIO *io, const char *file, int perm, int mode) {
```

Comment:
I suggest to add also `RZ_NONNULL` for `file` and `rz_return_val_if_fail(file, NULL)` since `strdup()` cannot accept NULL pointers